torch/csrc/jit/tensorexpr/operators/misc.cpp

```diff
@@ -642,6 +642,96 @@ Tensor computeCat(
       });
 }
 
+static bool checkStackInputShape(const std::vector<BufHandle>& bufList) {
+  TORCH_INTERNAL_ASSERT(
+      bufList.size() != 0,
+      buildErrorMessage("Empty input list is passed to aten::stack"));
+
+  auto buf0 = bufList.at(0);
+  TORCH_INTERNAL_ASSERT(
+      buf0.node()->dims().size() > 0, buildErrorMessage("Invalid buf rank"));
+  // Check if any of the dims is 0
+  bool hasEmptyDims = false;
+  std::vector<ExprHandle> dims;
+  for (const auto& dim : buf0.dims()) {
+    if (dim.AsNode<LongImm>() && immediateAs<int64_t>(dim) == 0ll) {
+      hasEmptyDims = true;
+    }
+    dims.push_back(IRSimplifier::simplify(dim));
+  }
+
+  // Check if all input bufs have the same size
+  for (int i = 1; i < bufList.size(); i++) {
+    auto buf = bufList.at(i);
+    TORCH_INTERNAL_ASSERT(
+        buf.node()->dims().size() == dims.size(),
+        buildErrorMessage("aten::stack inputs are with different sizes"));
+    auto dims_to_cmp = buf.dims();
+    for (int k = 0; k < dims.size(); k++) {
+      auto diff = IRSimplifier::simplify(
+          alloc<Sub>(dims_to_cmp.at(k).node(), dims.at(k).node()));
+      TORCH_INTERNAL_ASSERT(
+          diff->isConstant() && immediateAs<int>(diff) == 0,
+          buildErrorMessage("aten::stack inputs are with different sizes"));
+    }
+  }
+
+  return hasEmptyDims;
+}
+
+Tensor computeStack(
```

Comment:
Can we add an implementation that lowers into multiple loops like in the case of `aten::cat`, as done in `computeCatWoConditionals`?

The conditionals based lowering does not give good perf on CPUs, mainly because we can't vectorize it. I think the conditionals based lowering still makes sense for CUDA.

Reply:
> The conditionals based lowering does not give good perf on CPUs, mainly because we can't vectorize it. 

`aten::stack` only uses `compareSelect` conditionals which can be vectorized. I'll see if the other implementation has better performance. Will add it if it is.

Reply:
Please use some representative shapes from the models you have been working on, for perf analysis.

I still believe the multiple loops approach would be faster than conditionals because we will have both spacial and temporal cache locality for all inputs. It would be good to confirm with a perf analysis. 

If you want to do this as a followup, thats fine too.

Reply:
Sounds great! I'll first enable `aten::stack` in the fusion, and then do a perf analysis on real models for both approaches.